pygeoapi/provider/csv_.py

```diff
@@ -127,6 +136,25 @@ def _load(self, startindex=0, limit=10, resulttype='results',
                 feature_collection['numberMatched'] = \
                     len(feature_collection['features'])
 
+        if filter_expression:
```

Comment:
@jorgejesus : Json reponse is needed to let CQL filter logic work. So full json response logic has to be evaluated before `filter_expression`  check. And for `hits` to work in case where `filter_expression` is false, then json logic is not needed; `hits` can return the result without proceeding with json logic. But if `filter_expression` is true and `hits` is also true then json response is needed for getting proper count of results returned. Therefore such code checking and structure.

Reply:
please check this example which explains the work flow of above code:
![image](https://user-images.githubusercontent.com/59408650/89720583-2471b980-d9f1-11ea-8f6a-744929e82148.png)